crates/red_knot_python_semantic/src/types/diagnostic.rs

```diff
@@ -878,6 +879,24 @@ declare_lint! {
     }
 }
 
+declare_lint! {
+    /// ## What it does
+    /// Detects redundant `cast` calls where the value already has the target type.
+    ///
+    /// ## Why is this bad?
+    /// These casts have no effect and can be removed.
+    ///
+    /// ## Example
+    /// ```python
+    /// cast(int, 1)  # Redundant
+    /// ```
+    pub(crate) static REDUNDANT_CAST = {
+        summary: "detects redundant `cast` calls",
+        status: LintStatus::preview("1.0.0"),
+        default_level: Level::Error,
```

Comment:
Nit: This should probably be `Warn` by default
```suggestion
        default_level: Level::Warning,
```

Reply:
Thanks, i couldn't figure out how to assert a warning in the markdown tests.

Reply:
It should be the same as for errors. At least, that's what we do for other errors that have a warning severity by default